app/src/main/java/com/example/fittracker/activity/MainScreenActivity.java

```diff
@@ -41,4 +46,18 @@ public void onBackPressed() {
     public void onLogOutPressed() {
         presenter.onLogOutPressed();
     }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        switch (requestCode) {
```

Comment:
pass all this logic to the presenter pls

Reply:
Done